src/python/pants/pantsd/pants_daemon.py

```diff
@@ -319,8 +319,8 @@ def maybe_launch(self):
         self.terminate(include_watchman=False)
         self._logger.debug('launching pantsd')
         self.daemon_spawn()
-        # Wait up to 10 seconds for pantsd to write its pidfile so we can display the pid to the user.
-        self.await_pid(10)
+        # Wait up to 60 seconds for pantsd to write its pidfile.
+        self.await_pid(60)
```

Comment:
this is probably fine as a bandaid, but it ultimately means a 6x longer wait for legitimate failure cases. I'd be super shocked if it actually ever took even half that long (even under extreme load) - but I'm also surprised that it takes longer than 10s in some observed cases, mainly because theres no meaningful resource intensive work that happens on startup. I wonder if the main thread writing the pid is somehow contending with a fast spawning fs event handler thread?

at a minimum, a ticket/TODO would be good - there are a handful of strategies we could use to improve the spawn strategy here and it'd be good to further investigate the underlying apparent need for the lengthier wait now.

Reply:
At least as we observed in our internal ticket, the process did eventually start up (without a restart in the middle), so we haven't seen a legitimate failure case yet. And in fact, I expect that the longer timeout will do a better job of allowing us to distinguish between a slow machine and a legitimate failure. Right now it's somewhat ambiguous.

Reply:
So, essentially: if we do actually encounter legitimate failures we can loop back and examine the failure itself. Should I add a TODO to that effect?

Reply:
legitimate failure cases are more common during development, less so for users (we hope) but can still happen. the crux with this current logic tho is that if the daemon crashes immediately, it will now take 60 seconds vs 10 to do something w that because we're purely waiting on a file to appear. the TODO I'm thinking of would be more along the lines of improving the spawn+wait with something more robust to prevent a minute of waiting on something that may possibly never launch.

Reply:
k, forget it then.